DuggaSys/diagram.js

```diff
@@ -470,8 +481,17 @@ class StateMachine
         } else {
             console.error("Passed invalid argument to StateMachine.save() method. Must be a StateChange object!");
         }
+    };
+        settings.replay.timestamps = { 0: 0 }; // Clear the array with all timestamp.
+        this.historyLog.forEach(historyEntry => {
+            // settings.replay.timestamps[Object.keys(settings.replay.timestamps[Object.keys(settings.replay.timestamps).length - 1])]
+
+            if (settings.replay.timestamps[Object.keys(settings.replay.timestamps)[Object.keys(settings.replay.timestamps).length-1]] != historyEntry.time) {
```

Comment:
Please break this down.

Reply:
The one-liner has now been split apart to make it more readable.